### PR TITLE
Feature/improve settings migration

### PR DESCRIPTION
**Pull request by lasote**
Changelog: omit
Docs: omit

Close #5044

### CODE REVIEW

File: conans/test/functional/old/test_migrations.py [hunk 1]
```diff
@@ -17,6 +19,20 @@
...
 class TestMigrations(unittest.TestCase):
 
+    def is_there_var_for_settings_previous_version_test(self):
+        from conans import __version__ as current_version
+
+        tmp = Version(current_version)
+        if tmp.minor == 0:
+            return unittest.skip("2.0, this will make sense for 2.1")
+
+        previous_version = "{}.{}".format(tmp.major, int(tmp.minor) - 1)
```
Comment (lasote) [inline conans/test/functional/old/test_migrations.py]: Todo: improve with detection of previous patch too when patch > 0

File: conans/client/migrations.py [hunk 2]
```diff
@@ -25,33 +26,49 @@ def __init__(self, cache, current_version, out):
...
+            version_default_contents = getattr(migrations_settings, var_name)
+            if version_default_contents != default_settings_yml:
+                current_settings = load(self.cache.settings_path)
+                if current_settings != version_default_contents:
+                    save_new()
+                else:
+                    save(self.cache.settings_path, default_settings_yml)
             else:
-                save(self.cache.settings_path, default_settings_yml)
+                self.out.warn("Migration: Settings already up to date")
```
Comment (memsharded) [inline conans/client/migrations.py]: Why a warning?

**Comment by jgsogo** [comment]
@lasote, one doubt related to this PR:

The variable `default_settings_yml`, shouldn't it be equal to the last `settings_x_y_z` (which in turn should be `settings_<current_version>`?

**Comment by memsharded** [comment]
Yes, good question, that should remove some duplication, right? The bad thing is that suddenly, the default settings belong to the "migrations", which is a bit ugly regarding the structure of the codebase.

**Comment by jgsogo** [comment]
We can do it the other way around and add a test: the last line of the `migrations_***` could be `settings_<current_version> = default_settings_yml`; we can even create this variable dynamically and fail if someone has defined it, it would be complementary to the new test in this PR.

**Pull request merged**